tests/test_detect_file.py

```diff
@@ -26,7 +30,16 @@ def test_okay(self, filepath: str) -> None:
             f"Test case for {filepath!r} should NOT be vulnerable to dependency confusion"
 
     @pytest.mark.parametrize("filepath", list(_iter_files(os.path.join(Base.DATA_DIR, "requirements_files", "fail"))))
-    def test_fail(self, filepath: str) -> None:
+    def test_fail(self, filepath: str, index_allowlist: bool) -> None:
```

Comment:
Could `index_allowlist` be part of the `pytest.mark.parametrize`? Other tests use it, and it looks kind of weird to combine both. Also, `ALLOWED_INDEXES` could become part of the `pytest.mark.parametrize` as well.